shub/settings.py

```diff
@@ -668,4 +668,20 @@ def __iter__(self):
     }
 
 if ENABLE_GITLAB_AUTH:  # noqa
-    SOCIAL_AUTH_GITLAB_SCOPE = ["api", "read_user"]
+    if "SOCIAL_AUTH_GITLAB_SCOPE" not in locals():
```

Comment:
My preference would be to set this as a default empty list, and then check the environment (or other places) to update it, or set.

Reply:
I had tried that, but the yaml import code did not pull in the settings when i pre-populated the variable early in the code.

Reply:
I don't follow, sorry. The structure of settings.py is that we have defaults in these structures: https://github.com/singularityhub/sregistry/blob/5477171c8a042344fa2b01be97ba2cb534ba0e91/shub/settings.py#L28 and then we either update that from the settings file here: https://github.com/singularityhub/sregistry/blob/5477171c8a042344fa2b01be97ba2cb534ba0e91/shub/settings.py#L274-L282 - you should be able to follow a similar pattern here.

Reply:
right, but i could find no LIST_DEFAULTS, so i patterned it more how the PLUGINS_ENABLED is done.

Reply:
Maybe make `LIST_DEFAULTS` then, and (if it makes sense) use that for plugins too?

Reply:
should  the environment overwrite the settings.yaml and the secrets file, or should it append to the list and uniqify it.   currently PLUGINS_LIST appends what is in the env to the list, and i kinda copied that for the GITLAB scope, or should it be a full replacement?

Reply:
The order that we have currently starts with defaults, overrides with environment, and then checks for non None values in the settings.yaml. I would keep that for now, although looking at this again after many years typically you want environment variables to always take precedence. We can shuffle those around in another PR after this one if you agree.

Reply:
i did an overwrite as that matches what other things do.